Jenkinsfile

```diff
@@ -1,188 +1,80 @@
-          sleep 10
-          ./test/infrastructure/deployment.bash
-          docker-compose down --volumes --remove-orphans
-        """
-      }
-
-
-      stage("Publish docker images") {
-
-        sh "docker login -u ${env.DOCKER_USER} -p ${env.DOCKER_PASS}"
-        sh "docker tag scalableminds/webknossos:${env.BRANCH_NAME}__${env.BUILD_NUMBER} scalableminds/webknossos:${env.BRANCH_NAME}"
-        sh "docker push scalableminds/webknossos:${env.BRANCH_NAME}__${env.BUILD_NUMBER}"
-        sh "docker push scalableminds/webknossos:${env.BRANCH_NAME}"
-      }
-
-
-      stage("Build system packages") {
-
-        env.VERSION = readFile('version').trim()
-        sh "./buildtools/make_dist.sh oxalis ${env.BRANCH_NAME} ${env.BUILD_NUMBER}"
-
-        def base_port = 11000
-        def port = base_port + sh(returnStdout: true, script: """
-          grep -nx "${BRANCH_NAME}" /var/lib/jenkins/jobs/webknossos/branches.txt > /dev/null || \
-            echo "${BRANCH_NAME}" >> /var/lib/jenkins/jobs/webknossos/branches.txt
-          grep -nx "${BRANCH_NAME}" /var/lib/jenkins/jobs/webknossos/branches.txt | grep -Eo '^[^:]+' | head -n1
-          """).trim().toInteger()
-
-        def modes = ["dev", "prod"]
-        def pkg_types = ["deb", "rpm"]
-        for (int i = 0; i < modes.size(); i++) {
-          for (int j = 0; j < pkg_types.size(); j++) {
-            sh "./buildtools/build-helper.sh oxalis ${env.BRANCH_NAME} ${env.BUILD_NUMBER} ${port} ${modes[i]} ${pkg_types[j]}"
-          }
-        }
-
-        sh "mkdir packages && mv *.deb packages && mv *.rpm packages"
-      }
+    env.DOCKER_CACHE_PREFIX = "~/.webknossos-build-cache"
+    env.COMPOSE_PROJECT_NAME = "webknossos_${env.BRANCH_NAME}_${commit}"
+    sh "mkdir -p ${env.DOCKER_CACHE_PREFIX}"
+    sh "docker-compose pull sbt"
+  }
 
 
-      stage("Publish system packages") {
+  stage("Build") {
 
-        sh "echo ${env.BRANCH_NAME} > .git/REAL_BRANCH"
-        withEnv(["JOB_NAME=oxalis"]) {
-          sh "./buildtools/publish_deb.py"
-          sh "./buildtools/salt-redeploy-dev.sh"
-        }
-      }
+    sh "docker-compose run frontend-dependencies"
+    sh "docker-compose run frontend-docs"
+    sh "docker-compose run sbt clean compile stage"
+    sh "docker build -t scalableminds/webknossos:${env.BRANCH_NAME}__${env.BUILD_NUMBER} ."
+  }
 
 
-      currentBuild.result = "SUCCESS"
+  stage("Test") {
 
+    sh "docker-compose run frontend-linting"
+    sh "docker-compose run frontend-flow"
+    sh "docker-compose run frontend-tests"
+    // sh "docker-compose run e2e-tests"
+    sh """
+      DOCKER_TAG=${env.BRANCH_NAME}__${env.BUILD_NUMBER} docker-compose up webknossos &
+      sleep 10
+      ./test/infrastructure/deployment.bash
+      docker-compose down --volumes --remove-orphans
+    """
+  }
 
-    } catch (err) {
 
-      currentBuild.result = "FAILURE"
-      echo err.toString()
-      throw err
+  dockerPublish { repo = "scalableminds/webknossos" }
```

Comment:
This seems to misuse the closure, just to set a variable. I'm changing it to a normal function with a single parameter.

Reply:
Other libraries do use it that way, though.
https://github.com/fabric8io/jenkins-pipeline-library#deploy-project

Reply:
@normanrz Sorry, didn't see the comment in the outdated field.

Merged it already with the "normal" function that gets a map. I would argue for this version, as it contains the variable names as well (by accepting a map). Closures are fine if variables are dynamically set depending on the closure execution, but here I find normal functions much more intuitive.

If you prefer closures, we can also use them, but then also for `wrap` to have it consistent.

Reply:
I think you are right. We should stick with functions. Thanks for reviewing and merging this PR :-)